### PR TITLE
DTSPO-18260 - Auto bump template versions

### PR DESCRIPTION
**Pull request by endakelly**
### Jira link (if applicable)
https://tools.hmcts.net/jira/browse/DTSPO-18260

### Change description ###
Add a script and action to automatically bump template versions when changes are made so that new releases have completely new template versions

This will prevent issues occurring on AKS when newer versions of the chart are pulled

If there is an app with dependencies and those dependencies are both pulling chart-library on different versions, then the one downloaded last is used as helm has no native versioning system for the template files besides the naming. This causes multiple conflicts and it's tough to resolve.

The action will run two scripts. The first script will get the existing template versions from the master branch. The second script will loop over each changed file to get the templates that may need their version bumped and bump them by one.

Subsequent action runs won't continue to increment further. It will only ever bump it a single version above master.

The changes are then committed to the pull request branch.

Here is an image showing the script working via github actions:
Screenshot 2024-08-14 at 14 46 23

Readme updated with advice on updating downstream charts. Their checks should fail because the templates versions are bumped so new versions shouldn't get automerged.

### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: .github/workflows/bump-tpl-version.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        env:
+          CHANGED_FILES: ${{ steps.changed-files.outputs.all_changed_files }}
+
+      - name: Commit changes
+        if: env.SCRIPT_OUTPUT == 'version_updated'
+        run: |
+          git config user.name hmcts-platform-operations
+          git config user.email github-platform-operations@hmcts.net
+          git add .
+          git commit -m bump-version
```
Comment (reespozzi) [inline .github/workflows/bump-tpl-version.yaml:47]: Only a tiny thing but maybe the message could be a little more explicit like "Bump helm tpl versions" or something

**Review by reespozzi — COMMENTED** [review]
Looks great to me! Had a look through a few commits like https://github.com/hmcts/chart-library/pull/124/commits/4cc257f74801c7a0084f7aa2569cb96588f662d9 and seems to do exactly what we were after from the tech improvement


Only thing I'm a little unsure, I think we also need to update the definition versions of files which get bumped. i.E in this commit https://github.com/hmcts/chart-library/pull/124/commits/4cc257f74801c7a0084f7aa2569cb96588f662d9#diff-97f97fb0f0bd1d87ebb7d427930181f80a1596aa7990a08047e208f98094a4c2R1 the labels inclusion has been upped to v3, but we would still have two v2 configmap possibilities, each with different label definitions, if that makes sense?

**Comment by endakelly** [comment]
> 4cc257f#diff-97f97fb0f0bd1d87ebb7d427930181f80a1596aa7990a08047e208f98094a4c2R1

Ah I see what you mean, hmm, that could lead to cascading changes being made. Will have another look to see what would happen. Could well end up making more sense to simply increment everything per release. I did try initially to link the release version to the templates so it would be something like `hmcts.labels.v2.2.3` but the helm tests do not like that at all.

**Comment by endakelly** [comment]
> Looks great to me! Had a look through a few commits like [4cc257f](https://github.com/hmcts/chart-library/commit/4cc257f74801c7a0084f7aa2569cb96588f662d9) and seems to do exactly what we were after from the tech improvement
> 
> Only thing I'm a little unsure, do we also need to update the definition versions of files which get bumped. i.E in this commit [4cc257f#diff-97f97fb0f0bd1d87ebb7d427930181f80a1596aa7990a08047e208f98094a4c2R1](https://github.com/hmcts/chart-library/commit/4cc257f74801c7a0084f7aa2569cb96588f662d9#diff-97f97fb0f0bd1d87ebb7d427930181f80a1596aa7990a08047e208f98094a4c2R1) the labels inclusion has been upped to v3, but we would still have two v2 configmap possibilities, each with different label definitions, if that makes sense?

@reespozzi tried initially to trigger the action from the action so it would run multiple times but didn't like that approach. I've updated the script to loop over the files and it looks to be giving the desired results https://github.com/hmcts/chart-library/pull/124/commits/96b93a161fdb48375c2a6431ab6a744f64cefbc1

**Review by reespozzi — APPROVED** [review]
Nice work, **[example commit](https://github.com/hmcts/chart-library/commit/96b93a161fdb48375c2a6431ab6a744f64cefbc1)** seems to be exactly what we're after

**Pull request merged**